src/pyasm/web/app_server.py

```diff
@@ -250,21 +250,29 @@ def handle_not_logged_in(my, allow_change_admin=True):
                     if not current_project or current_project == "default":
                         current_project = Project.get_default_project()
                     if current_project and current_project != "default":
-                        Project.set_project(current_project)
+                        try:
+                            Project.set_project(current_project)
+                        except SecurityException, e:
+                            print e
+                            if 'is not permitted to view project' in e.__str__():
+                                web_wdg = None
+                            else:
+                                raise
 
```

Comment:
Where does web_wdg get defined to a value that isn't None?

It doesn't look like if web_wdg will ever return true.